src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.cs

```diff
@@ -0,0 +1,95 @@
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            Task<HttpResponseMessage> taskResponse = client.GetAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+            return GetFromJsonAsyncCore(taskResponse, type, options, cancellationToken);
+        }
+
+        public static Task<TValue> GetFromJsonAsync<TValue>(this HttpClient client, string? requestUri, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            Task<HttpResponseMessage> taskResponse = client.GetAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+            return GetFromJsonAsyncCore<TValue>(taskResponse, options, cancellationToken);
+        }
+
+        public static Task<TValue> GetFromJsonAsync<TValue>(this HttpClient client, Uri? requestUri, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            Task<HttpResponseMessage> taskResponse = client.GetAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+            return GetFromJsonAsyncCore<TValue>(taskResponse, options, cancellationToken);
+        }
+
+        public static Task<object?> GetFromJsonAsync(this HttpClient client, string? requestUri, Type type, CancellationToken cancellationToken = default)
+            => client.GetFromJsonAsync(requestUri, type, null, cancellationToken);
+
+        public static Task<object?> GetFromJsonAsync(this HttpClient client, Uri? requestUri, Type type, CancellationToken cancellationToken = default)
+            => client.GetFromJsonAsync(requestUri, type, null, cancellationToken);
+
+        public static Task<TValue> GetFromJsonAsync<TValue>(this HttpClient client, string? requestUri, CancellationToken cancellationToken = default)
+            => client.GetFromJsonAsync<TValue>(requestUri, null, cancellationToken);
+
+        public static Task<TValue> GetFromJsonAsync<TValue>(this HttpClient client, Uri? requestUri, CancellationToken cancellationToken = default)
+            => client.GetFromJsonAsync<TValue>(requestUri, null, cancellationToken);
+
+        private static async Task<object?> GetFromJsonAsyncCore(Task<HttpResponseMessage> taskResponse, Type type, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            using (HttpResponseMessage response = await taskResponse.ConfigureAwait(false))
+            {
+                response.EnsureSuccessStatusCode();
+                Debug.Assert(response.Content != null);
```

Comment:
Is it safe to assume that Content can't be null when the response originates from calling `GetAsync`?
@scalablecory @buyaa-n @stephentoub

Reply:
> Is it safe to assume that Content can't be null when the response originates from calling GetAsync?

Yes

Reply:
Well, technically, I'm wrong.  In practice that'll be the case, but it is possible to provide an HttpMessageHandler implementation where that's not the case.  So to be safe, this really should check for null.

Reply:
Ignore the 'confused' emoji. I don't know how to delete it.

Reply:
Based on that, I am doing a null-forgiving on this method and in case that `Content` is null, `response.Content.ReadFromJsonAsync` will throw `ArgumentNullException`.